adapter.go

```diff
@@ -388,28 +388,30 @@ func (a *Adapter) RemovePolicy(sec string, ptype string, rule []string) error {
 
 // AddPolicies adds multiple policy rules to the storage.
 func (a *Adapter) AddPolicies(sec string, ptype string, rules [][]string) error {
-	var err error
-	for _, rule := range rules {
-		line := a.savePolicyLine(ptype, rule)
-		err = a.db.Create(&line).Error
-		if err != nil {
-			return err
+	return a.db.Transaction(func(tx *gorm.DB) error {
+		for _, rule := range rules {
+			line := a.savePolicyLine(ptype, rule)
+			if err := tx.Create(&line).Error; err != nil {
+				tx.Rollback()
```

Comment:
According to the docs at https://gorm.io/docs/transactions.html#Transactions the returning of an err will rollback the transaction so no need to call tx.Rollback().

Reply:
@rwbarber2 fixed